widgetssdk/src/main/java/com/glia/widgets/di/GliaCoreImpl.kt

```diff
@@ -152,4 +156,10 @@ internal class GliaCoreImpl : GliaCore {
     override fun unsubscribeFromQueueUpdates(onError: Consumer<GliaException>?, callback: Consumer<Queue>) {
         Glia.unsubscribeFromQueueUpdates(onError, callback)
     }
+
+    override fun getCurrentVisitor(onResult: OnResult<Visitor>, onError: OnError) {
+        Glia.getCurrentVisitor { visitor, gliaException ->
+            visitor?.let(onResult::onResult) ?: onError.onError(gliaException.toWidgetsType("Failed to get current visitor"))
```

Comment:
I think the idea is that `GliaCoreImpl` works with Core types for exception and callbacks. Types are converted to Widgets types in `GliaWidgets` class. Have a look at how the other functions are implemented. 😉

Reply:
Yes! But what's wrong with this code?
I just use these callbacks because they fit my use case. The `RequestCallback` is quite uncomfortable to use.

Reply:
Let me think...

Reply:
Yes, the only concern is regarding consistency. Your approach is good but it's not consistent with the way how other functions in `GliaCoreImpl` are implemented. The matter is that we don't need to expose `getCurrentVisitor()` for integrators through `GliaWidgets`. That's the reason why this confusion appears. Otherwise you might add it to `GliaWidgets` and use it with Widgets callbacks.

Anyway, I would prefer to keep consistency in `GliaCoreImpl` although it's a bit inconvenient, agree. It's solely my opinion.

Reply:
`GliaCoreImpl` is internal.
But I think, I now understand what you mean. At least I can change the `onError` callback to not map it to Widgets type and then just ignore.

Reply:
Yes, it's internal, but it didn't have `OnResult`, `OnError`, `toWidgetsType` imports so far.

Reply:
fixed!